src/mscorlib/src/System/String.cs

```diff
@@ -573,10 +572,13 @@ public override bool Equals(Object obj)
         // Determines whether two strings match.
         [Pure]
         [ReliabilityContract(Consistency.WillNotCorruptState, Cer.MayFail)]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
@jkotas True, but if the JIT is able to inline the method then it may be able to delete some of the checks right? For example in the common case of

```cs
if (stringVar == "foo") { ... }
```

(which just calls `string.Equals`), it should be able to determine that `"foo"` (and possibly `stringVar`) isn't null and eliminate those checks. Would this really increase code bloat that much?